### ISSUE TITLE
Compilation Error: document must begin with a Setext-style header.

### ISSUE THREAD

**Issue by sbstnzmr** [leebyron/spec-md#21]
```
$ spec-md ./spec-md/project.md > ./spec-md/project.html
Error: Primary document D:\Windows\Desktop\markdown\spec-md\project.md is missing a title. It must begin with a Setext-style header. Example:

Title of Spec
=============

    at validateDocument (C:\Users\seb\AppData\Roaming\npm\node_modules\spec-md\src\parse.js:111:11)
    at Object.enter (C:\Users\seb\AppData\Roaming\npm\node_modules\spec-md\src\parse.js:90:9)
    at visit (C:\Users\seb\AppData\Roaming\npm\node_modules\spec-md\src\visit.js:95:26)
    at validate (C:\Users\seb\AppData\Roaming\npm\node_modules\spec-md\src\parse.js:87:10)
    at C:\Users\seb\AppData\Roaming\npm\node_modules\spec-md\src\parse.js:13:12
```

Even a document just containing

```
Title of Spec
=============
```

won't compile. Or from the documentation

```
Spec Markdown
-------------

Introductory paragraph.

# First Section Header
```

will deliver the same error.

**Comment by leebyron** [comment]
Sorry to hear you encountered problems with this tool.

I've tried to reproduce this issue and can't seem to. I added a test runner along with this case in the linked commit which shows a proper parse and print.

I'd be happy to re-open this issue and address a fix if you have an example of a specific document which is still failing with this error.

**Comment by csuwildcat** [comment]
I am experiencing the same exact issue - all combinations of the examples provided in the docs all produce the same error that there is no valid title, when there clearly is. In trying to find commonality between my situation and the other person who had this issue, may I ask if you are testing against Windows?

**Comment by csuwildcat** [comment]
For reference, I just ran the tool on your own documentation as a test, and it produced the following error:

```
Error: SyntaxError: spec.md:65:36
# [Contributing](./CONTRIBUTING.md)
                                   ^
Expected " ", "#", [# ] or [^\n\r# ] but end of input found.    at formatError 
```

**Comment by csuwildcat** [comment]
Update: I isolated that last bug I posted, and found it is caused by not having a return line under your last line of text.

After resolving that error, the tool returned to throwing the same Setext Title error I mentioned previously.

**Comment by leebyron** [comment]
I’m happy to reopen this if there’s a reproducible test case. It’s entirely possible that this is a Windows issue. The tests for this repo run on Linux.

**Comment by csuwildcat** [comment]
> I’m happy to reopen this if there’s a reproducible test case. It’s entirely possible that this is a Windows issue. The tests for this repo run on Linux.

I just used WSL to run it in Linux but it produced the same error. Not sure what to do about steps-to-reproduce, because I am literally just trying to parse both the simplest example of a near empty doc, as well as the various MDs in your repo. I really would love to use this tool, because I haven't seen anything that enables markdown use for specifications of this caliber.

**Comment by csuwildcat** [comment]
Just to be exhaustive, I cloned the actual spec-md repo and ran the `simple-header` test in both Windows and Linux (WSL Ubuntu) and still got the same error:

```
simple-header/input.md is missing a title. It must begin with a Setext-style header. Example:

Title of Spec
=============
```

**Comment by csuwildcat** [comment]
Further down the rabbit hole: I did a global install of the package, invoked the tool from the console *within* the global package, and to my surprise, that worked for rendering the test doc in the module. But the wild thing so far is that when I use the same globally installed module and invoke it from within my project folder, no dice, it throws the Setext Title error.

**Comment by csuwildcat** [comment]
OK, I think I have identified the issue. After stepping through grammar.js and testing/logging all the ops and calls inside of `parsetitle()`. Something looked a bit off with some of the substr() returns, namely, it would print what looked to be the same string, then check equality to whatever the next step's value is. Visually they looked the same, but after I checked their lengths, one was always being returned with an additional char: '=== ' vs '==='.

When the code hits this line in grammar.js: `if (input.substr(peg$currPos, 3) === peg$c7)`, I believe it should be finding them equal, but one of the strings has a hanging space character. When I modified that one bit of code to this: `if (input.substr(peg$currPos, 4).trim() === peg$c7)`, the file is parsed/generated just about right, except individual lines in paragraphs are being separated into their own paragraphs. I am not sure if that is a separate bug, or related to the change I made above.
